pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -176,6 +176,10 @@ func (s *akvKMProvider) GetKeys(ctx context.Context) (map[keymanagementprovider.
 			return nil, nil, fmt.Errorf("failed to get key objectName:%s, objectVersion:%s, error: %w", keyVaultKey.Name, keyVaultKey.Version, err)
 		}
 
+		if keyBundle.Attributes != nil && keyBundle.Attributes.Enabled != nil && !*keyBundle.Attributes.Enabled {
+			return nil, nil, fmt.Errorf("key %s version %s is disabled. please re-enable in azure key vault or remove reference to this key", keyVaultKey.Name, keyVaultKey.Version)
```

Comment:
just wonder if should just skip it or return err in this case

Reply:
yes ideally we should skip and warn the user. Unfortunately, right now we don't have a mechanism to inform the users on a specific key/certificate failure. (this ties into the discussion we had at the PR meeting on Monday). I'm thinking for now we stay consistent with other behavior and throw error and then when we introduce more granular error handling we update?

Reply:
yeah, that makes sense to me.